src/middlewared/middlewared/plugins/vm.py

```diff
@@ -535,10 +533,7 @@ def identify_hypervisor(self):
         return False
 
     async def _extend_vm(self, vm):
-        vm['devices'] = []
-        for device in await self.middleware.call('vm.device.query', [('vm', '=', vm['id'])]):
-            device.pop('vm', None)
-            vm['devices'].append(device)
+        vm['devices'] = await self.middleware.call('vm.device.query', [('vm', '=', vm['id'])])
```

Comment:
Why no longer remove vm attribute?

Reply:
On vm update, we now require the user to pass vm attribute as a must so as to explicitly tell that the vm device in question belongs to `vm` id. This makes the transition helpful for the user who can retrieve list of devices attached to the vm and then if some aren't desired, pop them out and send the updated list back.

Reply:
I dont understand, why does it have to be in the query result? thats redundant

Reply:
It is redundant and doesn't help the user in any way except when he/she wants to update the vm and some devices. In that case they can just make amends and pass the list to middlewared, else they would have to add vm attribute to each device list. Please let me know if you think we should change this behavior

Reply:
Well, I guess it wont hurt having a bit redundancy then.